symfony/webpack-encore-pack/1.0/webpack.config.js

```diff
@@ -4,21 +4,23 @@ Encore
     // the project directory where compiled assets will be stored
     .setOutputPath('public/build/')
     // the public path used by the web server to access the previous directory
-    .setPublicPath('/build')
+    .setPublicPath('build/')
```

Comment:
This triggers:
> WARNING   The value passed to setPublicPath() should *usually* start with "/" or be a full URL (http://...). If you're not sure, then you should probably change your public path and make this message disappear.

but this is the only way to make assets path-independent. Using the `asset()` function adds the prefix in Symfony.

Reply:
I think `/build` IS correct. But, even with `/build`, the "keys" in `manifest.json` should be relative - it will generate something like this:

```json
{
  "build/0.b76febcc.js": "/build/0.b76febcc.js",
  "build/admin.css": "/build/admin.fd4fe345.css",
  "build/admin.js": "/build/admin.eb9c5c59.js",
}
```

The keys will never have the opening `/`, which is why you can say `asset('build/admin.js')` and it matches.

Reply:
When deploying an app in a subdirectory (like I just did), it breaks...

Reply:
but if you make it relative, things will break for any path not going through the Asset component, as they would be relative to the current page (which may be nested deeper than your root). and there might be a bunch of cases:

- JS imports handled by webpack when using code splitting (`import()` or the older `require.ensure()`)
- image paths referenced from JS (requiring an image in webpack works, and returns the path for this image in the build, including any versioning)
- image (and font) paths referenced from CSS may be an issue as well, I'm not sure. Webpack *could* generate relative paths there (as such paths are relative to the CSS location, not to the document), but it could as well be using absolute paths to keep things simpler for them

We **cannot** ensure that all assets are going through the Symfony component when dealing with more complex frontend code referencing assets from the frontend too.

Reply:
Do we need some integration then? It feels odd to have to hardcode the subpath in a webpack config when the Symfony app is just fine wherever we put it.

Reply:
Which kind of integration ? the Asset component guesses the subfolder being used based on the incoming request. The webpack builds is finished since ages at this point.

Reply:
Any integration that would make it seamless to put a Symfony app in a subdirectory.

Reply:
next solution: give up on any webpack feature referencing an asset from another one... Not really an option once you embrace webpack

Reply:
understood, updated